docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,418 @@
+    },
+    // ...other config
+}
+```
+
+Use built-in config, local custom config, and custom config from an npm package:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    extends: ["eslint:recommended", "./custom-config.js", "eslint-config-my-config"],
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+#### Flat Config Examples
+
+Use recommended config:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customConfig from "./custom-config.js";
+import myConfig from "eslint-config-my-config";
+
+export default [
+    js.configs.recommended,
+    customConfig,
+    myConfig,
+    {
+        rules: {
+            semi: ["warn", "always"]
+        },
+        // ...other config
+    }
+];
+```
+
+Note that because you are just importing JavaScript modules, you can mutate the config objects before exporting them. For example, you might want to have a certain config object only apply to your test files:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customTestConfig from "./custom-test-config.js";
+
+export default [
+    js.configs.recommended,
+    {
+        ...customTestConfig,
+        files: ["**/*.test.js"],
+    },
+];
+```
+
+### Ignoring Files
+
+With eslintrc, you can make ESLint ignore files by creating a separate `.eslintignore` file in the root of your project. The `.eslintignore` file uses the same glob pattern syntax as `.gitignore` files. Alternatively, you can use an `ignorePatterns` property in your eslintrc file.
+
+To ignore files with flat config, you can use the `ignores` property in a config object. The `ignores` property accepts an array of glob patterns. Note that flat config glob patterns do _not_ match dot files (e.g. `.dotfile.js`).
+
+#### eslintrc Examples
+
+`.eslintignore` example:
+
+```shell
+# .eslintignore
+temp.js
+.config/*
+# ...other ignored files
+```
+
+`ignorePatterns` example:
+
+```javascript
+// .eslintrc.js
+{
+    // ...other config
+    ignorePatterns: ["temp.js", ".config/*"],
+}
+```
+
+#### Flat Config Examples
+
+```javascript
+export default [
+    {
+        // ...other config
+        ignores: ["temp.js", ".config/*"]
+    }
+];
+```
+
+### CLI Flag Changes
+
+The following CLI flags are no longer supported with the flat config file format:
+
+* `--rulesdir`
+* `--ext`
+* `--resolve-plugins-relative-to`
+
+The flag `--no-eslintrc` has been replaced with `--no-config-lookup`.
+
+### Additional Changes
+
+The following changes have been made from the eslintrc to the flat config file format:
+
+* The `root` option no longer exists.
+* The `files` option cannot be a single string anymore, it must be an array.
+* The `sourceType` option now supports the new value `"commonjs"` (`.eslintrc` supports it too, but it was never documented).
+* You can configure `noInlineConfig` and `reportUnusedDisableDirectives` options under the setting `linterOptions`.
```

Comment:
I think we could use a section with an example for this instead of it just being a bullet point here.

Reply:
note to self: still have to implement

Reply:
implemented